src/transparent.rs

```diff
@@ -163,23 +165,61 @@ pub fn zstd_enable_transparent<'a>(ctx: &Context) -> anyhow::Result<ToSqlOutput<
     let table_name = &config.table;
     let new_table_name = format!("_{}_zstd", table_name);
 
-    let table_type: String = db
-        .query_row(
-            "select `type` from sqlite_master where name=?",
-            params![table_name],
-            |r| r.get(0),
-        )
-        .with_context(|| format!("Could not get table information for {}", table_name))?;
-    if table_type != "table" {
-        anyhow::bail!("{} is a {} not a table", table_name, table_type);
+    let configs = get_configs(&db)?;
+    let already_compressed_columns = configs
+        .iter()
+        .filter(|c| &c.table == table_name)
+        .map(|c| &c.column[..])
+        .collect::<Vec<&str>>();
+
+    log::debug!("already compressed columns={:?}", already_compressed_columns);
+
+    if already_compressed_columns.contains(&&config.column[..]) {
+        anyhow::bail!(
+            "Column {} is already enabled for compression.",
+            &config.column
+        );
     }
+
+    let table_already_enabled = !already_compressed_columns.is_empty();
+
+    let dict_id_columns: Vec<String> = if table_already_enabled {
+        let query = format!(
+            r#"select "from"
+            from pragma_foreign_key_list('{}')
+            where "table" = '_zstd_dicts'"#,
+            &new_table_name
+        );
```

Comment:
feels kinda weird to find these by foreign key, I'd probably have gotten them from already_compressed_columns? But I can't think of a specific reason why this is a bad idea, so I guess it's fine.

Reply:
Yes, it sounds good. But I believe quering db in this case is a more reliable way to get the current state of affairs:) in the sense of future features and refactoring